CMakeLists.txt

```diff
@@ -25,6 +25,39 @@ if(POLICY CMP0071)
   cmake_policy(SET CMP0071 NEW)
 endif()
 
+# Check if any relevant env vars were set from the build env scripts
+if(DEFINED ENV{X_VCPKG_APPLOCAL_DEPS_INSTALL} AND NOT DEFINED X_VCPKG_APPLOCAL_DEPS_INSTALL)
+  set(X_VCPKG_APPLOCAL_DEPS_INSTALL "$ENV{X_VCPKG_APPLOCAL_DEPS_INSTALL}" CACHE BOOL "")
+endif()
+
+if(DEFINED ENV{VCPKG_DEFAULT_TRIPLET} AND NOT DEFINED VCPKG_TARGET_TRIPLET)
+  set(VCPKG_TARGET_TRIPLET "$ENV{VCPKG_DEFAULT_TRIPLET}" CACHE STRING "")
+endif()
+
+if(DEFINED ENV{VCPKG_ROOT} AND NOT DEFINED CMAKE_TOOLCHAIN_FILE)
+  set(CMAKE_TOOLCHAIN_FILE "$ENV{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake" CACHE STRING "")
+endif()
+
+# Set a default build type if none was specified
+# See https://blog.kitware.com/cmake-and-the-default-build-type/ for details.
+set(default_build_type "Release")
+if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
+  # Debug builds are broken on Windows due to CMake/vcpkg linking issues.
+  # As a workaround, we just default to RelWithDebInfo on Windows.
+  if(WIN32)
+    set(default_build_type "RelWithDebInfo")
+  else()
+    set(default_build_type "Debug")
+  endif()
+endif()
+
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  message(STATUS "Setting CMAKE_BUILD_TYPE to '${default_build_type}' as none was specified.")
+  set(CMAKE_BUILD_TYPE "${default_build_type}" CACHE STRING "Choose the type of build." FORCE)
+  # Set the possible values of build type for cmake-gui
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
```

Comment:
MinSizeRel is not maintained, remove?

Reply:
It's one of the standard values supported by cmake. No idea if it works or not.

Reply:
It does not work, because it implies to use -Os, but we don't use the build type to set the optimization but the optimize parameter. "MinSizeRel" is not considered in this code. 

I just noticed that even "Release" is not proper handled everywhere. 
The later should be fixed because it is an often used build type. 
I can take care or will you? It is just a matter of check both types.

Reply:
Please do, you already looked into it. I suggest to merge this PR first, then follow up with more cleanup in another PR. At least this fixes the windows cli build.